### ISSUE TITLE
`void-function: directory-empty-p repo-dir` error on `doom install`

### ISSUE THREAD

**Issue by tiagommmelo** [doomemacs/doomemacs#7779]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

I'm using Ubuntu 22.04 LTS under Windows 11 WSL2.
After installing emacs and cloning hlissner repo on ~/.emacs.d, I cd into that and run `bin/doom install`. What is expected is that it'll install doom, however, it generates an error.
Two things to note:

1. I've installed it before on the same system and it worked; I've excluded it after playing for a while and was about to install it fresh (I've completed removed the .doom.d and .emacs.d directories after that, uninstalled emacs and then ran `sudo apt autoremove` as well); then I did all the process following the official instructions but it does not work anymore; perhaps I'm forgetting to remove something else before attempting to reinstall it all again?
2. I cannot fill in the system information as it is asked because doom actually does not install this time. Instead, what I can say is that:
- I'm running Windows 11 and WSL2 with Ubuntu 22.04 LTS
- the current installed emacs is **"GNU Emacs 27.1"** 

### Current behavior

When I run `bin/doom install`, it does not install and here's the error message:
```
❯ bin/doom install
Installing Doom Emacs!

- Skipping ~/.doom.d/ (already exists)
  - Skipping ~/.doom.d/init.el (already exists)...
  - Skipping ~/.doom.d/config.el (already exists)...
  - Skipping ~/.doom.d/packages.el (already exists)...
- Envvar file already exists, skipping
Installing plugins
> Ensuring packages are installed and built...
  x There was an unexpected runtime error
    Message: Symbol's function definition is void
    Details: (directory-empty-p)
    Backtrace:
      (directory-empty-p repo-dir)
      (or (not (file-directory-p repo-dir)) (directory-empty-p repo-dir))
      (while (or (not (file-directory-p repo-dir)) (directory-empty-p repo-dir)) (if (= try 3) (progn (error "Failed to clone package"))) (doom-print (doom-print--format "Failed to clone %S, trying again (attempt #%d)..." package (...
      (let ((try 0)) (while (or (not (file-directory-p repo-dir)) (directory-empty-p repo-dir)) (if (= try 3) (progn (error "Failed to clone package"))) (doom-print (doom-print--format "Failed to clone %S, trying again (attempt #%d...
      (let ((straight-vc-git-post-clone-hook (cons #'(lambda (&rest --cl-rest--) "\n\n(fn &key REPO-DIR COMMIT &allow-other-keys)" (let* ((repo-dir (car (cdr (plist-member --cl-rest-- ':repo-dir)))) (commit (car (cdr (plist-member ...
      (condition-case e (let ((straight-vc-git-post-clone-hook (cons #'(lambda (&rest --cl-rest--) "\n\n(fn &key REPO-DIR COMMIT &allow-other-keys)" (let* ((repo-dir (car (cdr (plist-member --cl-rest-- ':repo-dir)))) (commit (car (...
      (let ((repo-dir (straight--repos-dir (or local-repo package))) (build-dir (straight--build-dir package))) (if force-p nil (let* ((build (if (plist-member recipe :build) (plist-get recipe :build) t)) (want-byte-compile (or (eq...
      (let* ((--cl-rest-- (append (list :recipe recipe) recipe)) (package (car (cdr (plist-member --cl-rest-- ':package)))) (local-repo (car (cdr (plist-member --cl-rest-- ':local-repo)))) (recipe (car (cdr (plist-member --cl-rest-...
      (cl-destructuring-bind (&key package local-repo recipe &allow-other-keys) (append (list :recipe recipe) recipe) (let ((repo-dir (straight--repos-dir (or local-repo package))) (build-dir (straight--build-dir package))) (if for...
      (straight--with-plist (append (list :recipe recipe) recipe) (package local-repo recipe) (let ((repo-dir (straight--repos-dir (or local-repo package))) (build-dir (straight--build-dir package))) (if force-p nil (let* ((build (...
      (let ((recipe (car --dolist-tail--))) (straight--with-plist (append (list :recipe recipe) recipe) (package local-repo recipe) (let ((repo-dir (straight--repos-dir (or local-repo package))) (build-dir (straight--build-dir pack...
      (while --dolist-tail-- (let ((recipe (car --dolist-tail--))) (straight--with-plist (append (list :recipe recipe) recipe) (package local-repo recipe) (let ((repo-dir (straight--repos-dir (or local-repo package))) (build-dir (s...
    ! Wrote extended backtrace to ~/.emacs.d/.local/state/logs/cli.doom.240330091309.4445.error
```

### Steps to reproduce

1. Open bash
2. CD into ~/.emacs.d
3. Run `bin/doom install`
4. It generates the error reported above

### System Information

Explanation on why I'm not sending this above

**Comment by hlissner** [comment]
The problem is that the `directory-empty-p` function was introduced in Emacs 28.1. I'll fix this shortly, but I highly recommend you upgrade soon, because Doom will be dropping 27.x support in the near future.

**Comment by tiagommmelo** [comment]
Hello and thanks for the update. I've uninstalled 27.x actually and installed 28.1 already as well (clear installation) but the same error occurs. I find it is weird that it worked normally on 27.x, but, after I uninstalled it, this error happens not mattering the version of Emacs (apparently) I am.

Ah, I should also mention that I'm dropping it all (I'm preparing another computer for me to use) so it will be a clean install of Windows, WSL and Ubuntu and, therefore, it might just work normally when I try to install doom in the new WSL2 Ubuntu instance again. I'll go with the 28.1 directly this time to avoid issues and I hope this helps other people that eventually should fall into this issue.
